flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTestHarness.java

```diff
@@ -232,15 +231,18 @@ public Thread invoke() throws Exception {
 	 *
 	 */
 	public Thread invoke(StreamMockEnvironment mockEnv) throws Exception {
+		checkState(this.mockEnv == null);
+		checkState(this.taskThread == null);
 		this.mockEnv = checkNotNull(mockEnv);
 
 		initializeInputs();
 		initializeOutput();
 
-		this.task = taskFactory.apply(mockEnv);
-
-		taskThread = new TaskThread(task);
+		taskThread = new TaskThread(() -> taskFactory.apply(mockEnv));
```

Comment:
I'm missing the information why do we need this change. Could you explain more in the commit message and maybe optionally somewhere in the code as a comment?

Reply:
I've added a commit message (as commit description).